src/components/PostsTable.vue

```diff
@@ -1,104 +1,79 @@
 <template>
-  <div>
-    <table>
-      <thead>
-        <tr>
-          <th>Sr#</th>
-          <th>Author</th>
-          <th>Title</th>
-          <th>Actions</th>
-        </tr>
-      </thead>
-      <tbody>
-        <tr
-          v-for="(post, index) in posts"
-          :key="'post-row-' + post.data.created"
-        >
-          <td>{{ index + 1 }}</td>
-          <td>{{ post.data.author_fullname }}</td>
-          <td>{{ post.data.title }}</td>
-          <td><button>Detail</button></td>
-        </tr>
-      </tbody>
-    </table>
-  </div>
+  <ag-grid-vue
+    class="ag-theme-alpine"
+    :columnDefs="columnDefs"
+    :rowData="posts"
+    :frameworkComponents="frameworkComponents"
+    @grid-ready="onGridReady"
+  >
+  </ag-grid-vue>
 </template>
 
 <script>
+import { AgGridVue } from "ag-grid-vue";
+import BtnCellRenderer from "./btn-cell-renderer.vue";
+
 export default {
   name: "PostsTable",
+  components: {
+    AgGridVue,
+  },
+  data() {
+    return {
+      columnDefs: null,
+      rowData: null,
+      frameworkComponents: null,
+    };
+  },
   props: ["posts"],
+  methods: {
+    onGridReady(params) {
+      this.gridApi = params.api;
+      this.columnApi = params.columnApi;
+    },
+  },
+  beforeMount() {
+    this.rowData = this.posts;
+    this.columnDefs = [
```

Comment:
Can we move this initialization into another function and call the function in `data` property of vue?

Reply:
Done